### ISSUE TITLE
TeamSpeak ini and log files use wrong location when launched from another program

### ISSUE THREAD

**Issue by ghost** [IDI-Systems/acre2#467]
**Arma 3 Version:** `1.80` stable
**CBA Version:** `3.5` stable
**ACRE2 Version:** `2.5.1.980` stable

**Mods:**
```
- CBA_A3
- acre2
```

**Description:**
- I'm trying to understand what causes ACRE to place the ini and log file on the desktop so I can avoid it on clients (I am provisioning multiple clients and seeing it on all of them). Sorry for shortcutting some of the due diligence above but I just want to know the deciding factors for where ACRE will put these files and adjust if I can. Appreciate the help and if you do need some omitted details I'm happy to fill them in (If it depends on where I run arma from or something like that - that is enough detail to help me as I can change that). I know the ini ends up in the %APPDATA% area for teamspeak but, again, just trying to understand cause of it being on my desktop at any point.

**Steps to reproduce:**
- I'm running ACRE for the first time with Teamspeak
- Expected: ACRE doesn't place anything on the desktop
- Actual: ACRE places it's log and ini file on the desktop (just want to know why/cause)

**Where did the issue occur?**
- Pre-ARMA launch/ACRE init

**Placed Modules:**
- None

**RPT log file:**
- I can do this but I feel this is more of an understanding/documentation problem

**Comment by jonpas** [comment]
I assume those are TeamSpeak plugin ini and log files. I had the same last time, it seems to happen when launching TeamSpeak from some other program. I haven't looked into it yet.

Repurposing this issue as bug report.

**Comment by ghost** [comment]
That would make sense, I'm launching a batch file from Desktop and it's doing several things including starting teamspeak. I can give this a relatively quick test and update the ticket this afternoon

**Comment by ghost** [comment]
Yeah - it's definitely based on where I'm running teamspeak from, I can locally adjust that in some way though I'd rather teamspeak not do this to me :/

Thanks for the assistance, unless ACRE can do something about this you can feel free to close this (if you can do something please let me know and I will undo whatever dumb workaround I employ and can give it a shot)

**Comment by PabstMirror** [comment]
You might be able to use `start` to change the starting dir to the TS install
https://technet.microsoft.com/en-us/library/cc770297(v=ws.11).aspx

```
/d <Path> | Specifies the startup directory.
```

**Comment by jonpas** [comment]
Does that happen with any other plugin? I am not sure what even has logs like we do, but TS logs themselves don't get redirected, so I am assuming it's how we parse that path, relatively to where it is started instead of from TS location.

**Comment by ghost** [comment]
@PabstMirror - was already using start so I was switching to `/d` at the same time you were probably updating this ticket (which is what I needed to do)

@jonpas - I would assume you are inheriting the root process/container location and maybe not explicitly requesting/looking up path. I would have to browse the code more thoroughly to speak with any actual confirmation on that.

For the log this section ` acre2/extensions/src/ACRE2Shared/Log.cpp`
```
Log::Log(char *logFile) { 
    InitializeCriticalSection(&this->m_CriticalSection);

    if(logFile == NULL) {
        this->fileHandle = GetStdHandle(STD_OUTPUT_HANDLE);
        return;
    }

    this->fileHandle = CreateFileA(logFile, GENERIC_WRITE, FILE_SHARE_READ, NULL, OPEN_ALWAYS, 0, 0);
```

and this scream location inheritance ` acre2/extensions/src/ACRE2Core/Engine.cpp`
```
g_Log = (Log *)new Log("acre2.log");
```

**Comment by jonpas** [comment]
Indeed, we should change that to request path from TS or whatever. Thanks for the info! 👍

**Comment by jonpas** [comment]
This was fixed in #1110 by moving plugin log location to same location as Arma RPTs.
